sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/DownloadBlobToFileTest.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.perf;
+
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.perf.core.ContainerTest;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.UUID;
+
+import static com.azure.perf.test.core.TestDataCreationHelper.createRandomByteBufferFlux;
+
+public class DownloadBlobToFileTest extends ContainerTest<PerfStressOptions> {
+    private final BlobClient blobClient;
+    private final BlobAsyncClient blobAsyncClient;
+
+    private final File tempDir;
+
+    public DownloadBlobToFileTest(PerfStressOptions options) {
+        super(options);
+        String blobName = "downloadToFileTest";
+        blobClient = blobContainerClient.getBlobClient(blobName);
+        blobAsyncClient = blobContainerAsyncClient.getBlobAsyncClient(blobName);
+
+        try {
+            tempDir = Files.createTempDirectory("downloadToFileTest").toFile();
+            tempDir.deleteOnExit();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public Mono<Void> globalSetupAsync() {
+        return super.globalSetupAsync()
+            .then(blobAsyncClient.upload(createRandomByteBufferFlux(options.getSize()), null))
+            .then();
+    }
+
+    @Override
+    public void run() {
+        File file = new File(tempDir, UUID.randomUUID().toString());
+        try {
+            blobClient.downloadToFile(file.getAbsolutePath());
+        } finally {
+            // We don't use File.deleteOnExit.
+            // This would cause memory accumulation in the java.io.DeleteOnExitHook with random file names
+            // and eventually take whole heap in long-running benchmarks.
+            if (!file.delete()){
+                throw new IllegalStateException("Unable to delete test file");
+            }
+        }
+    }
+
+
+    @Override
+    public Mono<Void> runAsync() {
+        File file = new File(tempDir, UUID.randomUUID().toString());
+        return blobAsyncClient.downloadToFile(file.getAbsolutePath())
+            .doFinally(ignored -> {
+                // We don't use File.deleteOnExit.
+                // This would cause memory accumulation in the java.io.DeleteOnExitHook with random file names
+                // and eventually take whole heap in long-running benchmarks.
+                if (!file.delete()){
```

Comment:
Would the time required to "delete" the file after each download count towards the perf test op/s metrics?

Reply:
Yes. This will be ~constant component of results (relatively small though, it's just node removal from fs tree). So we'd be comparing (new_perf + const) against (old_perf + const).

The alternative of generating garbage and cleaning it up at the end (even with dir cleanup instead of shutdown hook per file) isn't great - for long running benchmarks with larger data size this might not scale well and eat up whole disk at some point.

Reply:
FWIW, some of the benchmarks that I'm running are expected to generate ~300GB of garbage.

Reply:
Yeah, delete operations are generally quick and may not change the perf numbers by a huge margin. However, we should consider having a `beforeRun()` and `afterRun()` methods in the perf framework that run before and after `runAsync()`/`run()` methods to allow for "setup-startTimer-runPerfTest-stopTimer-cleanup". 

cc: @g2vinay